src/main/java/kitchenpos/menus/tobe/domain/entity/Menu.java

```diff
@@ -47,65 +49,71 @@ public class Menu {
     )
     private List<MenuProduct> menuProducts;
 
-    @Transient
-    private UUID menuGroupId;
+    protected Menu() {
+    }
 
-    public Menu() {
+    public Menu(UUID id, MenuName name, MenuPrice price, MenuGroup menuGroup, boolean displayed,
+                List<MenuProduct> menuProducts) {
+        this.id = id;
+        this.name = name.getValue();
+        this.price = price.getValue();
+        this.menuGroup = menuGroup;
+        this.displayed = displayed;
+        this.menuProducts = menuProducts;
     }
 
     public UUID getId() {
         return id;
     }
 
-    public void setId(final UUID id) {
-        this.id = id;
-    }
-
     public String getName() {
         return name;
     }
 
-    public void setName(final String name) {
-        this.name = name;
-    }
-
     public BigDecimal getPrice() {
         return price;
     }
 
-    public void setPrice(final BigDecimal price) {
-        this.price = price;
+    public void changePrice(final MenuPrice menuPrice, ProductRepository productRepository) {
+        BigDecimal threshHoldPrice = this.getSumOfProductPriceAndQuantity();
+        if (menuPrice.getValue().compareTo(threshHoldPrice) > 0) {
+            throw new IllegalArgumentException("New Price cannot be greater than threshHoldPrice");
+        }
+        this.price = menuPrice.getValue();
     }
 
     public MenuGroup getMenuGroup() {
         return menuGroup;
     }
 
-    public void setMenuGroup(final MenuGroup menuGroup) {
-        this.menuGroup = menuGroup;
-    }
-
     public boolean isDisplayed() {
         return displayed;
     }
 
-    public void setDisplayed(final boolean displayed) {
-        this.displayed = displayed;
-    }
-
-    public List<MenuProduct> getMenuProducts() {
-        return menuProducts;
+    public void displayOn(ProductRepository productRepository) {
+        BigDecimal threshHoldPrice = getSumOfProductPriceAndQuantity();
+        if (this.price.compareTo(threshHoldPrice) > 0) {
+            throw new IllegalStateException();
+        }
+        this.displayed = true;
     }
 
-    public void setMenuProducts(final List<MenuProduct> menuProducts) {
-        this.menuProducts = menuProducts;
+    public void hide() {
+        this.displayed = false;
     }
 
-    public UUID getMenuGroupId() {
-        return menuGroupId;
+    public List<MenuProduct> getMenuProducts() {
+        return menuProducts;
     }
 
-    public void setMenuGroupId(final UUID menuGroupId) {
-        this.menuGroupId = menuGroupId;
+    public BigDecimal getSumOfProductPriceAndQuantity() {
+        BigDecimal sum = BigDecimal.ZERO;
+        for (final MenuProduct menuProduct : this.menuProducts) {
+            sum = sum.add(
+                menuProduct.getProductPrice()
+                           .multiply(BigDecimal.valueOf(menuProduct.getQuantity()))
```

Comment:
아래와 같이 menuProduct 객체에게 메시지를 보내볼 수 있겠네요. (네이밍은 그냥 생각나는대로 적은거라 참고정도만 해주세요. 😅)
```suggestion
                menuProduct.calculatePrice()
```

Reply:
적용했습니다!